File: src/data/courses/dsa/subarrays-subsets-recursion/recursion-fundamentals/recursion-tree-understanding/code/java/RecursionTreeUnderstanding.java
class Recursion {
    // Fibonacci recursion tree
    int fib(int n) {
        if (n <= 1) return n;
        return fib(n-1) + fib(n-2);
    }
    
    // Print recursion tree
    void printTree(int n, String indent) {
        System.out.println(indent + "fib(" + n + ")");
        if (n > 1) {
            printTree(n-1, indent + "  ");
            printTree(n-2, indent + "  ");
        }
    }
}

File: src/data/courses/dsa/subarrays-subsets-recursion/recursion-fundamentals/recursion-tree-understanding/code/python/recursion-tree-understanding.py
def fib(n):
    if n <= 1:
        return n
    return fib(n-1) + fib(n-2)

def print_tree(n, indent=""):
    print(f"{indent}fib({n})")
    if n > 1:
        print_tree(n-1, indent + "  ")
        print_tree(n-2, indent + "  ")